flow/connectors/postgres/validate.go

```diff
@@ -112,6 +112,53 @@ func (c *PostgresConnector) CheckReplicationPermissions(ctx context.Context, use
 	return nil
 }
 
+func (c *PostgresConnector) CheckPublicationPermission(ctx context.Context, tableNameString string) error {
+	publication := "_PEERDB_DUMMY_PUBLICATION_" + shared.RandomString(4)
+	// check and enable publish_via_partition_root
+	supportsPubViaRoot, _, err := c.MajorVersionCheck(ctx, POSTGRES_13)
+	if err != nil {
+		return fmt.Errorf("error checking Postgres version: %w", err)
+	}
+	var pubViaRootString string
+	if supportsPubViaRoot {
+		pubViaRootString = "WITH(publish_via_partition_root=true)"
+	}
+	tx, err := c.conn.Begin(ctx)
+	if err != nil {
+		return fmt.Errorf("error starting transaction: %w", err)
+	}
+	defer tx.Conn().Close(ctx)
```

Comment:
Don't do this, connection should be closed when connector is closed, as is connector breaks after calling this method